src/shims/linux_stubs.h

```diff
@@ -16,6 +16,16 @@
 #ifndef __DISPATCH__STUBS__INTERNAL
 #define __DISPATCH__STUBS__INTERNAL
 
+#if !defined(TAILQ_FOREACH_SAFE)
+#define TAILQ_FOREACH_SAFE(var, head, field, temp)                             \
+  for ((var) = TAILQ_FIRST((head));                                            \
+       (var) && ((temp) = TAILQ_NEXT((var), field), 1); (var) = (temp))
+#endif
+
+#if !defined(TRASHIT)
```

Comment:
Just following the android pattern, where the `android_stubs.h` always define it.